### PR TITLE
Clarify some examples and change order in Pattern docs

### PR DESCRIPTION
**Pull request by MasonProtter**
Hi, I know I promised a long time ago that I'd try to help out with some docs, well here I am!

I basically am trying to add a bit of explanation in words before the code to try and clarify what various patterns and their names mean / do. I changed the order of things a little bit because I thought this made a little more sense. 

Feel free to only accept some of these changes (or none of them) or suggest tweaks to convey the meaning you want to convey.

Hope this is helpful!

### CODE REVIEW

File: docs/syntax/pattern.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ Patterns provide convenient ways to manipulate data.
 Literal Patterns
 ------------------------
 
+Patterns with a literal (e.g. `1`, `false`, `nothing`, `[4.0, 1f-6]`, etc) on the left hand side will check if the the argument is equal to that literal:
```
Comment (thautwarm) [inline docs/syntax/pattern.md:9]: `[4.0, 1f-6]` is actually an array pattern

Comment (MasonProtter) [reply docs/syntax/pattern.md:9]: Oops, I was thinking about how people call that an "array literal" in julia, but this is not what you mean in this package by literal.

Comment (MasonProtter) [reply docs/syntax/pattern.md]: Fixed.

**Comment by MasonProtter** [comment]
Do we need to do something special to trigger redeployment of the docs? It looks like the changes haven't taken effect here: https://thautwarm.github.io/MLStyle.jl/latest/syntax/pattern.html

**Comment by thautwarm** [comment]
Yes, by `bash build-docs.sh`, but it seems that it can only be properly executed by me.. 
In the summer vacation, I can find time to integrate the documentation build with travis ci.

**Comment by thautwarm** [comment]
done

**Pull request merged**